haystack/utils/import_utils.py

```diff
@@ -5,12 +5,48 @@
 import zipfile
 import requests
 import logging
+import importlib
 from pathlib import Path
 
 
 logger = logging.getLogger(__name__)
 
 
+def safe_import(import_path, classname, dep_group):
+    """
+    Method that allows the import of "non-existing" document stores.
+    Doc stores can be installed one by one with extras_require (see setup.cfg)
+    but they need to be all imported in haystack.document_stores.__init__()
+    
+    Therefore, in case of an ImportError, the class to import is replaced by
+    a hollow MissingDependency function, which will throw an error when
+    inizialized.
+    """
+    try:
+        module = importlib.import_module(f"{import_path}")
+        classs = vars(module).get(classname)
+    except ImportError as ie:
+        classs = _missing_dependency_stub_factory(classname, dep_group, ie)
+    return classs
+
+
+def _missing_dependency_stub_factory(classname, dep_group, import_error):
+    """
+    Create custom versions of MissingDependency using the given parameters.
+    See `safe_import()`
+    """
+    class MissingDependency:
+
+        def __init__(self, *args, **kwargs):
+            raise ImportError(f"Failed to import {classname}. "
+                            f"Run 'pip install farm-haystack[{dep_group}]' to fix this error.") from import_error
```

Comment:
I'd make the error message here (and everywhere else) even more descriptive and clarify that this is not an unexpected error but just due to missing an optional dependency. How about something like this: 

``` Failed to import {classname}, which is an optional component in Haystack. Install all required dependencies for it via `'pip install farm-haystack[{dep_group}]' to make the component available"